### ISSUE TITLE
No module named `griffe.agents.extensions`

### ISSUE THREAD

**Issue by MiguelGuthridge** [mkdocstrings/mkdocstrings#600]
**Please open an issue on [Griffe](https://github.com/mkdocstrings/griffe/issues) (new Python handler)
or [pytkdocs](https://github.com/mkdocstrings/pytkdocs/issues) (legacy Python handler) instead
if this is related to Python docstrings parsing or the collection of Python objects!**

**Describe the bug**
I recently updated my dependencies. When I try to build my project's documentation, I get an error.

<details>

<summary>Click to show traceback</summary>

```py
Error reading page 'arrangement.md': No module named 'mkdocstrings.handlers.python'
Traceback (most recent call last):
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib64/python3.11/site-packages/mkdocstrings/handlers/base.py", line 563, in get_handler
    module = importlib.import_module(f"mkdocstrings_handlers.{name}")
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib64/python3.11/site-packages/mkdocstrings_handlers/python/__init__.py", line 3, in <module>
    from mkdocstrings_handlers.python.handler import get_handler
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib64/python3.11/site-packages/mkdocstrings_handlers/python/handler.py", line 14, in <module>
    from griffe.agents.extensions import load_extensions
ModuleNotFoundError: No module named 'griffe.agents.extensions'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/migue/Source/FL-Studio-API-Stubs/scripts/build_docs.py", line 13, in <module>
    build()
  File "/home/migue/Source/FL-Studio-API-Stubs/scripts/build_docs.py", line 9, in build
    mkdocs_build(config)
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib/python3.11/site-packages/mkdocs/commands/build.py", line 322, in build
    _populate_page(file.page, config, files, dirty)
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib/python3.11/site-packages/mkdocs/commands/build.py", line 175, in _populate_page
    page.render(config, files)
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib/python3.11/site-packages/mkdocs/structure/pages.py", line 271, in render
    self.content = md.convert(self.markdown)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib/python3.11/site-packages/markdown/core.py", line 254, in convert
    root = self.parser.parseDocument(self.lines).getroot()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib/python3.11/site-packages/markdown/blockparser.py", line 84, in parseDocument
    self.parseChunk(self.root, '\n'.join(lines))
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib/python3.11/site-packages/markdown/blockparser.py", line 99, in parseChunk
    self.parseBlocks(parent, text.split('\n\n'))
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib/python3.11/site-packages/markdown/blockparser.py", line 117, in parseBlocks
    if processor.run(parent, blocks) is not False:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib64/python3.11/site-packages/mkdocstrings/extension.py", line 121, in run
    html, handler, data = self._process_block(identifier, block, heading_level)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib64/python3.11/site-packages/mkdocstrings/extension.py", line 176, in _process_block
    handler = self._handlers.get_handler(handler_name, handler_config)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/migue/.cache/pypoetry/virtualenvs/fl-studio-api-stubs-oz04xZPt-py3.11/lib64/python3.11/site-packages/mkdocstrings/handlers/base.py", line 565, in get_handler
    module = importlib.import_module(f"mkdocstrings.handlers.{name}")
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mkdocstrings.handlers.python'
```

</details>

While this initially appeared to be a duplicate of #454, I am already specifying the `python` extra for `mkdocstrings`, and have verified that it is installed correctly.

I am using the following dependency logic for documentation in my project. For the full dependencies, [here is a link](https://github.com/MiguelGuthridge/FL-Studio-API-Stubs/blob/20fa34d2a57d55c6863386ca02caa92c9303ae25/pyproject.toml#L68).

```toml
mkdocs = "^1.4.2"
mkdocs-gen-files = "^0.4.0"
mkdocs-autorefs = "^0.4.1"
mkdocs-material = "^8.5.11"
mkdocs-material-extensions = "^1.1.1"
mkdocstrings = {version = "^0.19.0", extras = ["python"]}
```

Here are some relevant sections from my `poetry.lock` file.

```toml
[[package]]
name = "griffe"
version = "0.32.3"
description = "Signatures for entire Python programs. Extract the structure, the frame, the skeleton of your project, to generate API documentation or find breaking changes in your API."
category = "dev"
optional = false
python-versions = ">=3.8"
files = [
    {file = "griffe-0.32.3-py3-none-any.whl", hash = "sha256:d9471934225818bf8f309822f70451cc6abb4b24e59e0bb27402a45f9412510f"},
    {file = "griffe-0.32.3.tar.gz", hash = "sha256:14983896ad581f59d5ad7b6c9261ff12bdaa905acccc1129341d13e545da8521"},
]

[package.dependencies]
colorama = ">=0.4"
```

```toml
[[package]]
name = "mkdocstrings"
version = "0.19.1"
description = "Automatic documentation from sources, for MkDocs."
category = "dev"
optional = false
python-versions = ">=3.7"
files = [
    {file = "mkdocstrings-0.19.1-py3-none-any.whl", hash = "sha256:32a38d88f67f65b264184ea71290f9332db750d189dea4200cbbe408d304c261"},
    {file = "mkdocstrings-0.19.1.tar.gz", hash = "sha256:d1037cacb4b522c1e8c164ed5d00d724a82e49dcee0af80db8fb67b384faeef9"},
]

[package.dependencies]
Jinja2 = ">=2.11.1"
Markdown = ">=3.3"
MarkupSafe = ">=1.1"
mkdocs = ">=1.2"
mkdocs-autorefs = ">=0.3.1"
mkdocstrings-python = {version = ">=0.5.2", optional = true, markers = "extra == \"python\""}
pymdown-extensions = ">=6.3"
```

**To Reproduce**

1. Clone the project at [this commit](https://github.com/MiguelGuthridge/FL-Studio-API-Stubs/tree/20fa34d2a57d55c6863386ca02caa92c9303ae25).
2. Install the project using `poetry install`
3. Attempt to build the documentation using `poetry run python -m scripts.build_docs`

**Expected behavior**
The documentation builds correctly as it does in the main branch.

**Information (please complete the following information):**
- OS: Fedora Linux 38
- `mkdocstrings` version: see dependencies listed above

**Additional context**
From a Python shell (`poetry run python`)  I was able to verify that the dependencies were installed using import statements, but importing the specific submodules caused errors.

```py
⋙   poetry run python
Python 3.11.4 (main, Jun  7 2023, 00:00:00) [GCC 13.1.1 20230511 (Red Hat 13.1.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import griffe
>>> import griffe.agents.extensions
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'griffe.agents.extensions'
>>> 
>>> import mkdocstrings
>>> import mkdocstrings.handlers.python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'mkdocstrings.handlers.python'
>>> 
```

**Comment by pawamoy** [comment]
Hey, thanks for the report. Can you try to upgrade `mkdocstrings-python`? What your installed version of it?

**Comment by MiguelGuthridge** [comment]
Thanks so much for the pointer, I figured it out!

The installed version of `mkdocstrings-python` was `0.8.3`, which was significantly outdated. I presume it stayed that way because that was valid according to the constraints of `mkdocstrings == 0.19.1`. 

Due to a misconfiguration on my end, it was stuck at that version because I had some non-primary dependencies (`mkdocs-autorefs` and `mkdocs-material-extensions`) in my `pyproject.toml` which were left over from back when the project used `requirements.txt` and there was no distinction between primary and secondary dependencies.

After removing the unneeded packages from my dependency list, I was able to fix the issue by updating to `mkdocstrings == 0.20.0` which updated `mydocstrings-python` to `1.3.0`, fixing the issue.
